subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/ProjectDetails.java

```diff
@@ -19,12 +19,15 @@
 import org.gradle.api.Project;
 
 import javax.annotation.Nullable;
+import java.util.Objects;
 
 public interface ProjectDetails {
 
     static ProjectDetails of(final Project project) {
         final String displayName = project.getDisplayName();
         final String description = project.getDescription();
+        final String name = project.getName();
```

Comment:
🤔 This will make existing uses of `ProjectDetails` require more data to be captured than before, right? I'm not sure consolidating usages of `Project` data in `ProjectDetails` is the best path here, compared to defining separate models for the different tasks needs.

Reply:
I considered that... since the additional payload seemed small enough, I leaned towards reusing it. These are current usages now btw:
![image](https://user-images.githubusercontent.com/2187522/210371358-6684d62b-8354-453c-a568-c7008e8c969e.png)

Reply:
@bamboo I am still leaning towards using the same data structure used elsewhere in the report plugins for these tasks. 

Is this good to go in your view?

Reply:
@bamboo PTAL: dd46120c6f9aaff4a800c9763a5df7ccf4ed6a15 

This commit uses multiple ProjectDetails variations that contain the data that is required by each task (only HTML Dependency Report task uses the new one which contains project name and path).

Reply:
Nice! 👍